src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -302,23 +303,40 @@ protected virtual string DiscoverServiceName()
 			return null;
 		}
 
-		internal static string AdaptServiceName(string originalName) => originalName?.Replace('.', '_');
+		internal static string AdaptServiceName(string originalName)
+		{
+			var serviceNameRegex = new Regex("^[a-zA-Z0-9 _-]+$");
+			var adaptedName = originalName?.Replace('.', '_');
+
+			if (adaptedName != null && serviceNameRegex.IsMatch(adaptedName))
+			{
+				return adaptedName;
+			}
+
+			return null;
```

Comment:
So returning `null` would not be necessary anymore.

Reply:
fixed